test/unit_tests/models/api_calls_test.dart

```diff
@@ -0,0 +1,159 @@
+  final jsonDetailedDevice = """
+    {
+    "kind": "kind",
+    "etag": "etag",
+    "deviceId": "device ID",
+    "serialNumber": "serial number",
+    "status": "status",
+    "lastSync": "last sync",
+    "supportEndDate": "support end date",
+    "annotatedUser": "user",
+    "annotatedLocation": "location",
+    "annotatedAssetId": "asset ID",
+    "notes": "notes",
+    "model": "model",
+    "meid": "meid",
+    "orderNumber": "order number",
+    "willAutoRenew": true,
+    "osVersion": "os version",
+    "platformVersion": "plataform version",
+    "firmwareVersion": "firmware version",
+    "macAddress": "mac address",
+    "bootMode": "boot mode",
+    "lastEnrollmentTime": "last enrolled date",
+    "orgUnitPath": "org unit path"
+    }""";
+
+  /// Group of tests to do to the api calls, each test has its objective
+  group('Api Calls', () {
+    test(
+        'returns AccountDevices if the http call completes successfully without next page token',
+        () async {
+      final client = MockClient();
+      when(client.get(ApiCalls.DEVICE_LIST_URL,
+              headers: {HttpHeaders.authorizationHeader: 'Bearer $authToken'}))
+          .thenAnswer((_) async => http.Response(jsonAccountDevices, 200));
+      final apiCall = ApiCalls(client);
+
+      expect(
+          await apiCall.getDeviceList(authToken, null), isA<AccountDevices>());
+    });
+
+    test(
+        "throws an exception if the http call for list of devices without next page token completes with an error",
+        () {
+      final client = MockClient();
+      when(client.get(ApiCalls.DEVICE_LIST_URL,
+              headers: {HttpHeaders.authorizationHeader: 'Bearer $authToken'}))
+          .thenAnswer((_) async => http.Response('Not Found', 404));
+      final apiCall = ApiCalls(client);
+
+      expect(apiCall.getDeviceList(authToken, null), throwsException);
+    });
+
+    test(
+        'returns AccountDevices if the http call completes successfully with next page token',
+        () async {
+      final client = MockClient();
+      when(client.get(ApiCalls.DEVICE_LIST_WITH_TOKEN_URL + nextPageToken,
+              headers: {HttpHeaders.authorizationHeader: 'Bearer $authToken'}))
+          .thenAnswer((_) async => http.Response(jsonAccountDevices, 200));
+      final apiCall = ApiCalls(client);
+
+      expect(await apiCall.getDeviceList(authToken, nextPageToken),
+          isA<AccountDevices>());
+    });
+    test(
+        "throws an exception if the http call for list of devices with next page token completes with an error",
+        () {
+      final client = MockClient();
+      when(client.get(ApiCalls.DEVICE_LIST_WITH_TOKEN_URL + nextPageToken,
+              headers: {HttpHeaders.authorizationHeader: 'Bearer $authToken'}))
+          .thenAnswer((_) async => http.Response('Not Found', 404));
+      final apiCall = ApiCalls(client);
+
+      expect(apiCall.getDeviceList(authToken, nextPageToken), throwsException);
```

Comment:
Is there a way to compare the error code in  the test?

Reply:
With the Error Handler #35 I will, by the moment it just caught the exception.